src/platform/site-wide/user-nav/containers/AutoSSO.jsx

```diff
@@ -1,23 +1,37 @@
+import React from 'react';
 import { connect } from 'react-redux';
 
 import { checkKeepAlive } from 'platform/user/authentication/actions';
 import {
   ssoeInbound,
   hasCheckedKeepAlive,
+  isAuthenticatedWithSSOe,
 } from 'platform/user/authentication/selectors';
+import { isProfileLoading } from 'platform/user/selectors';
 import { hasSession } from 'platform/user/profile/utilities';
 import { checkAutoSession } from 'platform/utilities/sso';
 import { removeLoginAttempted } from 'platform/utilities/sso/loginAttempted';
 
 function AutoSSO(props) {
-  const { useInboundSSOe, hasCalledKeepAlive } = props;
+  const {
+    useInboundSSOe,
+    hasCalledKeepAlive,
+    authenticatedWithSSOe,
+    profileLoading,
+  } = props;
 
   if (hasSession()) {
     removeLoginAttempted();
   }
 
-  if (useInboundSSOe && !hasCalledKeepAlive) {
-    checkAutoSession().then(() => {
+  if (
+    // avoid race condition where hasSession hasn't been set
+    !window.location.pathname.includes('auth/login/callback') &&
```

Comment:
should we do the same thing with the `/logout/` page?

Reply:
I don't think the logout page includes the user-nav, but I will double check

Reply:
It does. I think it wouldn't get that far, cuz the page redirects super fast, and I don't think it includes calls to the feature toggles, but I can add that check here just to be doubly certain